src/qt/radiobut.cpp

```diff
@@ -39,11 +42,50 @@ bool wxRadioButton::Create( wxWindow *parent,
              const wxString& name)
 {
     m_qtRadioButton = new QRadioButton( parent->GetHandle() );
+
+    if ( (style & wxRB_GROUP) || (style & wxRB_SINGLE) )
+    {
+        JoinNewGroup();
+
+        if( style & wxRB_SINGLE )
+        {
+            // Ensure that other buttons cannot join this group
+            m_lastGroup.erase(parent);
+        }
+        else
+        {
+            m_lastGroup[parent] = m_qtButtonGroup;
+            m_qtRadioButton->setChecked(true); // The first button in a group should be selected
+        }
+    }
+    else
+    {
+        // Add it to the previous group, if any
+        std::map<wxWindow*, QButtonGroup*>::iterator it = m_lastGroup.find(parent);
+        if ( it != m_lastGroup.end() )
+        {
+            it->second->addButton(m_qtRadioButton);
+        }
+    }
+
     m_qtRadioButton->setText( wxQtConvertString( label ));
 
     return QtCreateControl( parent, id, pos, size, style, validator, name );
 }
 
+wxRadioButton::~wxRadioButton()
+{
+    if ( m_qtRadioButton->group() &&
+         m_qtRadioButton->group()->buttons().size() == 1 )
+    {
```

Comment:
The comment for this chunk of code would definitely be easier to read than the code itself.